### ISSUE TITLE
Why createResultValue assert result == Result::eSuccess when exception is disabled?

### ISSUE THREAD

**Issue by huisedenanhai** [KhronosGroup/Vulkan-Hpp#644]
I found the following code in vulkan.hpp

```c++
  template <typename T>
  VULKAN_HPP_INLINE typename ResultValueType<T>::type createResultValue( Result result, T & data, char const * message )
  {
#ifdef VULKAN_HPP_NO_EXCEPTIONS
    ignore(message);
    VULKAN_HPP_ASSERT( result == Result::eSuccess ); // WHY ???
    return ResultValue<T>( result, std::move( data ) );
#else
    if ( result != Result::eSuccess )
    {
      throwResultException( result, message );
    }
    return std::move( data );
#endif
  }
```

That's quite confusing as checking the return code should be done by the caller.

**Comment by mocabe** [comment]
Looks like a bug. You can submit PR or I'll work on this.

**Comment by huisedenanhai** [comment]
@mocabe  It will be great help if you can fix that.

Maybe a flag can be added for users to disable those assertions in `createResultValue`. It's  sometimes helpful to have the library check the return code.

**Comment by mocabe** [comment]
> Maybe a flag can be added for users to disable those assertions in createResultValue. It's sometimes helpful to have the library check the return code.

It would be helpful in some situations, but I don't think asserting return value of underlying API is not something to be done by default (considering "NoException" mode is basically "I do all error handling by myself" mode). Making that opt-in still makes sense, though.

**Comment by asuessenbach** [comment]
Agreed, providing an opt-out option for the VULKAN_HPP_ASSERT in the createResultValue family of functions might be worth to have.
Something like VULKAN_HPP_ASSERT_ON_RESULT, which maps to VULKAN_HPP_ASSERT when it's not explicitly defined by the user.

**Comment by mocabe** [comment]
Ok. I will submit PR to support opt-out.

**Comment by asuessenbach** [comment]
@mocabe Would you mind to add some short docu about that new macro to the readme.md?

**Comment by asuessenbach** [comment]
Resolved by #649.

**Comment by dsvensson** [comment]
Stumbled across this when trying to figure out why Queue::submit, Queue::submit2KHR, Queue::submit2 all are void even if I have VULKAN_HPP_NO_EXCEPTIONS. Shouldn't the result be propagated?
